mosquitto/docker-compose.yaml

```diff
@@ -0,0 +1,18 @@
+services:
+  app_proxy:
+    environment:
+      APP_HOST: mosquitto_broker_1
+      APP_PORT: 9001
+    container_name: mosquitto_app_proxy_1
+  broker:
+    image: eclipse-mosquitto:2.0.20@sha256:8b396cec28cd5e8e1a3aba1d9abdbddd42c454c80f703e77c1bec56e152fa54e
+    restart: on-failure
+    ports:
+      - '1883:1883'
+      - '9001:9001'
+    volumes:
+      - ${APP_DATA_DIR}/config:/mosquitto/config:rw
+      - ${APP_DATA_DIR}/data:/mosquitto/data:rw
+      - ${APP_DATA_DIR}/log:/mosquitto/log:rw
+      - ${APP_DATA_DIR}/www:/mosquitto/www:r
```

Comment:
To keep everything under one data directory like other apps I'd suggest going with something like this:

```
data
├── mosquitto-data
│   └── .gitkeep
├── config
│   └── mosquitto.conf.default
└── log
    └── .gitkeep
```

And then if serving the UI from a Docker container as described above, just get rid of `- ${APP_DATA_DIR}/www:/mosquitto/www:r`